datafusion/src/row/reader.rs

```diff
@@ -174,14 +184,22 @@ impl<'a> RowReader<'a> {
 
     #[inline(always)]
     fn null_bits(&self) -> &[u8] {
-        let start = self.base_offset;
-        &self.data[start..start + self.null_width]
+        if self.null_free {
+            &[]
+        } else {
+            let start = self.base_offset;
+            &self.data[start..start + self.null_width]
```

Comment:
if `null_width` is always zero, I wonder if the check for `self.null_free` is needed?